### PR TITLE
fix: allow ppapi processes access to resource bundle on all platforms

### LINKED ISSUE #29326: [Bug]: PDF viewer displays blank pages on Windows
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

13.0.1

### What operating system are you using?

Windows

### Operating System Version

Windows 10 20H2

### What arch are you using?

x64

### Last Known Working Electron version

_No response_

### Expected Behavior

PDF viewer should display content of the PDF file.

### Actual Behavior

image

All PDFs are affected.

Interestingly PDF viewer displays number of pages correctly so it seems to read the PDF file, but then fails to render it.

There does not seem to be anything of interest in the console.

This bug seems to affect Windows specifically, on Ubuntu the PDFs are rendered correctly.

### Testcase Gist URL

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by VerteDinde**
#### Description of Change

Fixes #29326

~The chromium roll #28462 included a change to shell/browser/extensions/electron_extension_system.cc to address https://chromium-review.googlesource.com/c/chromium/src/+/2771320. This change also included moving away from base::string types and toward using UTFString helpers. Because of these changes, Windows now needs access to the shared resource bundle in order to use the GetStringFUTF16 helper method for PDF metadata. Specifically, here: https://source.chromium.org/chromium/chromium/src/+/main:ui/base/l10n/l10n_util.cc;l=809;bpv=1~

This chromium roll #28462 changed the PDF viewer so that file size is now included among the properties sent in the PDF metadata. To accomplish that, the ppapi subprocesses are now allowed access to the resource bundle on all platforms (previously, they were gated to be Mac-only): https://chromium-review.googlesource.com/c/chromium/src/+/2619003

This PR updates shell/app/electron_main_delegate to allow ppapi subprocesses access to the resource bundle and align with chrome_main_delegate.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue with PDF loading on Windows, where metadata was not loading correctly and causing the PDF to load as a blank page.

### CODE REVIEW

File: shell/app/electron_main_delegate.cc [hunk 1]
```diff
@@ -90,10 +90,13 @@ bool SubprocessNeedsResourceBundle(const std::string& process_type) {
...
-      // Mac needs them too for scrollbar related images and for sandbox
-      // profiles.
-      process_type == ::switches::kPpapiPluginProcess ||
+      // Mac needs them for sandbox profiles.
       process_type == ::switches::kGpuProcess ||
+#endif
+#if defined(OS_MAC) || defined(OS_WIN)
+      // Windows needs ppapi for loading pdf metadata, Mac needs it
+      // scrollbar related images
+      process_type == ::switches::kPpapiPluginProcess ||
```
Comment (deepak1556) [inline shell/app/electron_main_delegate.cc:99]: Would be good to align this with https://chromium-review.googlesource.com/c/chromium/src/+/2619003/9/chrome/app/chrome_main_delegate.cc

**Comment by deepak1556** [comment]
The linked CL only seem to switch to mojo for ipc, I don't see how resource bundle is involved because of it, can you clarify. Thanks! I believe the actual crash is caused due to https://chromium-review.googlesource.com/c/chromium/src/+/2619003 ?

File: shell/app/electron_main_delegate.cc (hunk 1 above)
Comment (VerteDinde) [reply shell/app/electron_main_delegate.cc:99]: Oh, great catch! I'll move ppapi process to all platforms to align with Chrome, will test and make sure everything works as expected 🙂

**Comment by VerteDinde** [comment]
Thanks @deepak1556! You're right, I was wrong - I couldn't find a root cause and was tracing back from the base::string changes and mojom conversations that broke the PDF viewer before in 13. The CR you linked is absolutely the root cause 🙂 Will adjust my PR description!

**Review by deepak1556 — APPROVED** [review]
👍

**Pull request merged**